### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by stephnna**
Hi code reviewer,

I fixed the N+1 problem and implement an integration test for the user and post view

Kindly review as necessary 
Thanks!

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe 'User index', type: :feature do
+  before(:each) do
```
Comment (SajjadAhmad14) [inline spec/features/user_index_spec.rb:4]: Please add a test that redirects to the user's show page when the user clicks on a user.

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe 'User show', type: :feature do
+  before(:each) do
```
Comment (SajjadAhmad14) [inline spec/features/user_show_spec.rb:4]: Please add a test that redirects to the post's show page when the user clicks on a user's post.

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index', type: :feature do
```
Comment (SajjadAhmad14) [inline spec/features/post_index_spec.rb:3]: Please add a test that asserts the user can see a section for pagination if there are more posts than fit on the view.
Please add a test that redirects to the post's show page when the user clicks on a post.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Stephen, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/user_index_spec.rb (hunk 1 above)
Comment (stephnna) [reply spec/features/user_index_spec.rb:4]: Great, done. Thanks!

File: spec/features/user_show_spec.rb (hunk 2 above)
Comment (stephnna) [reply spec/features/user_show_spec.rb:4]: Nice, done, thanks!

File: spec/features/post_index_spec.rb (hunk 3 above)
Comment (stephnna) [reply spec/features/post_index_spec.rb:3]: Done, thanks!

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @stephnna , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**